src/main/java/com/nourish1709/project3_main_service/CategoryController.java

```diff
@@ -0,0 +1,56 @@
+package com.nourish1709.project3_main_service;
+
+import com.nourish1709.project3_main_service.models.dto.CategoryDto;
+import com.nourish1709.project3_main_service.services.CategoryService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/categories")
+public class CategoryController {
+
+    private final CategoryService categoryService;
+
+    @PostMapping(
+            consumes = MediaType.APPLICATION_JSON_VALUE,
+            produces = MediaType.APPLICATION_JSON_VALUE
+    )
+    public ResponseEntity<CategoryDto> addCategory(@RequestBody CategoryDto categoryDto) {
+        CategoryDto categoryDto1 = categoryService.create(categoryDto);
+        return new ResponseEntity<>(categoryDto1, HttpStatus.CREATED);
+    }
+
+    @GetMapping(produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<List<CategoryDto>> getAllCategories() {
+        return new ResponseEntity<>(categoryService.getAll(), HttpStatus.OK);
+    }
+
+    @GetMapping(value = "/{id}", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<CategoryDto> getCategoryById(@PathVariable Long id) {
+        CategoryDto categoryDto = categoryService.getById(id);
+        if (categoryDto == null) {
```

Comment:
in fact we don't need this if statement here because even if we don't find category by provided id, exception will be thrown in CategoryService